.github/workflows/ci.yml

```diff
@@ -0,0 +1,129 @@
+# This workflow will build a .NET project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-net
+
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+    - main
+  workflow_dispatch:
+
+permissions:
+  pull-requests: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  DOTNET_CLI_PERF_LOG: false
+  DOTNET_CLI_TELEMETRY_OPTOUT: true
+  DOTNET_GENERATE_ASPNET_CERTIFICATE: false
+  DOTNET_NOLOGO: true
+  SuppressNETCoreSdkPreviewMessage: true
+  SLN_PATH: source/FlashOWare.Generators.sln
+
+jobs:
+  build:
+    name: Restore & Build & Test
+    continue-on-error: false
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        configuration: [Debug, Release]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 1
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        global-json-file: global.json
```

Comment:
We use `SLN_PATH` as an environment variable, and others as scoped variables (`COVERAGE_OUTPUT_PATH`, ...).
Should we have all other files checked into the repository as environment variables as well? And also non generated directories, like `source/gen` for _mutation testing_?
(`global.json`, `nuget.config`, `dotnet-tools.json`, `stryker-config.yaml`, ...)
But then I would not do it for workflow-generated files ... but perhaps workflow-generated files only within the job/step scope in which they are used ... what do you think?